### PR TITLE
* Dump 0x20 bytes of stack memory when dumping the register content.

### PR DESCRIPTION
**Pull request by RobertBuhren**

### CODE REVIEW

File: psp-core.c [hunk 1]
```diff
@@ -704,4 +704,28 @@ void PSPEmuCoreStateDump(PSPCORE hCore)
         else
             fprintf(stderr, "Querying CPU mode failed with %d\n", pspEmuCoreErrConvertFromUcErr(rcUc));
     }
+
+    /* Dump last 0x20 bytes of stack memory */
+    size_t cwLen = 8;
+    uint32_t awStack[cwLen];
+    rc = PSPEmuCoreMemRead(hCore, au32Reg[PSPCOREREG_SP] - cwLen, awStack, sizeof(awStack));
```
Comment (AlexanderEichner) [inline psp-core.c]: The address calculation "au32Reg[PSPCOREREG_SP] - cwLen" looks wrong, it should be "au32Reg[PSPCOREREG_SP] - cwLen * sizeof(uint32_t)" most likely. I'd prefer the following for reading the stack:

uint32_t au32Stack[8];
rc = PSPEmuCoreMemRead(hCore, au32Reg[PSPCOREREG_SP] - sizeof(au32Stack), &au32Stack[0], sizeof(au32Stack));
if (!rc)
...

File: psp-core.c [hunk 2]
```diff
@@ -704,4 +704,28 @@ void PSPEmuCoreStateDump(PSPCORE hCore)
...
+
```
Comment (AlexanderEichner) [inline psp-core.c]: Blank line can be deleted

File: psp-core.c [hunk 3]
```diff
@@ -704,4 +704,28 @@ void PSPEmuCoreStateDump(PSPCORE hCore)
...
+      printf( "Stack:\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x\n"
+              "\t0x%08x: 0x%08x <= SP\n",
+              au32Reg[PSPCOREREG_SP] -0x1c, awStack[0], au32Reg[PSPCOREREG_SP] -0x18, awStack[1],
```
Comment (AlexanderEichner) [inline psp-core.c]: Space between - and 0x1c etc., renaming to au32Stack

File: psp-core.c [hunk 4]
```diff
@@ -704,4 +704,28 @@ void PSPEmuCoreStateDump(PSPCORE hCore)
...
     }
+
+    /* Dump last 0x20 bytes of stack memory */
+    size_t cwLen = 8;
+    uint32_t awStack[cwLen];
+    rc = PSPEmuCoreMemRead(hCore, au32Reg[PSPCOREREG_SP] - cwLen, awStack, sizeof(awStack));
+
+    if (!rc)
+    {
+      printf( "Stack:\n"
```
Comment (AlexanderEichner) [inline psp-core.c:713]: Proper 4 spaces indentation please :P

**Review by AlexanderEichner — CHANGES_REQUESTED** [review]

**Review by RobertBuhren — COMMENTED** [review]
:+1:

**Comment by AlexanderEichner** [comment]
Merged, thanks!

**Pull request merged**